server/src/test/java/org/opentosca/toscana/plugins/lifecycle/LifecycleAwarePluginTest.java

```diff
@@ -0,0 +1,106 @@
+package org.opentosca.toscana.plugins.lifecycle;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.opentosca.toscana.core.BaseJUnitTest;
+import org.opentosca.toscana.core.testdata.TestPlugins;
+import org.opentosca.toscana.core.transformation.TransformationContext;
+import org.opentosca.toscana.core.transformation.platform.Platform;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class LifecycleAwarePluginTest extends BaseJUnitTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(LifecycleAwarePluginTest.class);
+
+    @Mock
+    public TransformationContext context;
+
+    @Mock
+    public TransformationLifecycle lifecycle;
+
+    public LifecycleTestPlugin plugin;
+
+    @Before
+    public void setUp() throws Exception {
+        when(context.getLogger((Class<?>) any(Class.class))).thenReturn(LoggerFactory.getLogger("Dummy Logger"));
+        plugin = new LifecycleTestPlugin(TestPlugins.PLATFORM1);
+    }
+
+    @Test
+    public void checkEnvFailure() throws Exception {
+        plugin.envCheckReturnValue = false;
+        try {
+            plugin.transform(context);
+        } catch (ValidationFailureException e) {
+            logger.info("Thrown exception: ", e);
+            verify(lifecycle, never()).checkModel();
+            return;
+        }
+        fail();
+    }
+
+    @Test
+    public void checkModelFailure() throws Exception {
+        try {
+            plugin.transform(context);
+        } catch (ValidationFailureException e) {
+            logger.info("Thrown exception: ", e);
+            verify(lifecycle, never()).prepare();
+            return;
+        }
+        fail();
+    }
+
+    @Test
+    public void successfulExec() throws Exception {
+        when(lifecycle.checkModel()).thenReturn(true);
+        plugin.transform(context);
+        verify(lifecycle).cleanup();
+    }
+
+    @Test(expected = Exception.class)
+    public void exceptionDuringTransformation() throws Exception {
+        when(lifecycle.checkModel()).thenReturn(true);
+        doThrow(new Exception("Test Exception")).when(lifecycle).transform();
```

Comment:
You are testing the exception mechanism of java..? I'm sure it works without bugs!

Reply:
Well this test was intended to check that the implementation of the `transform` method is re-throwing the execption as expected.

The test obviously is senseless, because the exception gets passed through by java.